### PR TITLE
working in main, not worikng in tests:(

### PR DESCRIPTION
**Pull request by krzeelzb**

### CODE REVIEW

File: krzeelzb/assignment-4/.idea/compiler.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (imorlowska) [inline krzeelzb/assignment-4/.idea/compiler.xml]: you can use .gitignore to specify you don't want this folder to be uploaded.

File: krzeelzb/assignment-4/src/main/java/Island.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import java.util.List;
+
+public class Island {
+
+    int columnsNumber = WorldMap.getColumnSize();
+    int rowsNumber = WorldMap.getRowSize();
+    int numberOfIslands = 0;
+    boolean[][] visited = new boolean[rowsNumber][columnsNumber];
+
+    private HashMap<String, List<Integer>> moves = new HashMap<String, List<Integer>>() {{
```
Comment (imorlowska) [inline krzeelzb/assignment-4/src/main/java/Island.java]: Do we really need the String in the HashMap? Do you think using a HashSet would be better here?

Comment (imorlowska) [inline krzeelzb/assignment-4/src/main/java/Island.java]: How about the generic approach with grids other than a simple square grid?

File: krzeelzb/assignment-4/src/main/java/WorldMapRepository.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        ArrayList<String> row2 = new ArrayList<String>(Arrays.asList("Land", "Land", "Sea", "Sea"));
+        ArrayList<String> row3 = new ArrayList<String>(Arrays.asList("Sea", "Sea", "Land", "Sea"));
+        ArrayList<String> row4 = new ArrayList<String>(Arrays.asList("Sea", "Sea", "Land", "Sea"));
+        ArrayList<ArrayList<String>> mapOne = new ArrayList<ArrayList<String>>(Arrays.asList(row1, row2, row3, row4));
+        WorldMap map = new WorldMap();
+        map.setMap(mapOne);
+        return map;
+    }
+
+    public static WorldMap getMapTwo(){
```
Comment (imorlowska) [inline krzeelzb/assignment-4/src/main/java/WorldMapRepository.java]: getOnlySeaMap?

File: krzeelzb/assignment-4/src/main/java/WorldMapRepository.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        ArrayList<String> row1 = new ArrayList<String>(Arrays.asList("Sea", "Sea", "Sea", "Sea"));
+        ArrayList<String> row2 = new ArrayList<String>(Arrays.asList("Sea", "Sea", "Sea", "Sea"));
+        ArrayList<String> row3 = new ArrayList<String>(Arrays.asList("Sea", "Sea", "Sea", "Sea"));
+        ArrayList<String> row4 = new ArrayList<String>(Arrays.asList("Sea", "Sea", "Sea", "Sea"));
+        ArrayList<ArrayList<String>> mapOne = new ArrayList<ArrayList<String>>(Arrays.asList(row1, row2, row3, row4));
+        WorldMap map = new WorldMap();
+        map.setMap(mapOne);
+        return map;
+    }
+    public static WorldMap getMapThree(){
```
Comment (imorlowska) [inline krzeelzb/assignment-4/src/main/java/WorldMapRepository.java]: getLandOnlyMap?

File: krzeelzb/assignment-4/src/main/java/WorldMapRepository.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class WorldMapRepository {
```
Comment (imorlowska) [inline krzeelzb/assignment-4/src/main/java/WorldMapRepository.java]: How about adding a few more cases? e.g.
S L
L L 

or 

S L S
L L L
S L S

or 

L S S
S L S
S S L

File: krzeelzb/assignment-4/.idea/compiler.xml (hunk 1 above)
Comment (krzeelzb) [reply krzeelzb/assignment-4/.idea/compiler.xml]: yes I will add gitignore

File: krzeelzb/assignment-4/src/main/java/Island.java (hunk 2 above)
Comment (krzeelzb) [reply krzeelzb/assignment-4/src/main/java/Island.java]: I will change it to HashSet. I used HashMap to make it clearer what are my moves, but I guess I can also add it in docs.

File: krzeelzb/assignment-4/src/main/java/WorldMapRepository.java (hunk 3 above)
Comment (krzeelzb) [reply krzeelzb/assignment-4/src/main/java/WorldMapRepository.java]: yes, way better

File: krzeelzb/assignment-4/src/main/java/WorldMapRepository.java (hunk 4 above)
Comment (krzeelzb) [reply krzeelzb/assignment-4/src/main/java/WorldMapRepository.java]: better:)

File: krzeelzb/assignment-4/src/main/java/WorldMapRepository.java (hunk 5 above)
Comment (krzeelzb) [reply krzeelzb/assignment-4/src/main/java/WorldMapRepository.java]: added:)

**Comment by krzeelzb** [comment]
I have a problem with Tests, because when I check the results in the Main class the results are correct, however when I run my test cases the results of calling `island.findIslands(map)` is always 3, and I don't know why.

File: krzeelzb/assignment-4/src/main/java/WorldMap.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import java.util.ArrayList;
+
+public class WorldMap {
+    static ArrayList<ArrayList<String>> map;
```
Comment (imorlowska) [inline krzeelzb/assignment-4/src/main/java/WorldMap.java]: Why is this static? The map is specific to the instance - keeping this static pollutes all the objects to have only one current map in memory - this is why your tests fail. :)

File: krzeelzb/assignment-4/src/main/java/Island.java [hunk 7]
```diff
@@ -0,0 +1,73 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+
+public class Island {
+
+    int columnsNumber = WorldMap.getColumnSize();
```
Comment (imorlowska) [inline krzeelzb/assignment-4/src/main/java/Island.java]: Do not use the static state! The map size will be based on the instance of the worldMap.

File: krzeelzb/assignment-4/src/main/java/Island.java [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+     */
+    private HashSet<List<Integer>> moves = new HashSet<List<Integer>>() {{
+        add(new ArrayList<Integer>(Arrays.asList(-1, 0)));
+        add(new ArrayList<Integer>(Arrays.asList(0, -1)));
+        add(new ArrayList<Integer>(Arrays.asList(0, 1)));
+        add(new ArrayList<Integer>(Arrays.asList(1, 0)));
+    }};
+
+
+    private void findIsland(WorldMap map, boolean[][] visited, int row, int column, int number) {
```
Comment (imorlowska) [inline krzeelzb/assignment-4/src/main/java/Island.java]: You need to update the column and row number here based on the map

File: krzeelzb/assignment-4/src/main/java/Island.java [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+        /**
+         recursive function that searches for the "land" tiles in a WorldMap
```
Comment (imorlowska) [inline krzeelzb/assignment-4/src/main/java/Island.java]: Nice javadoc - but please move it above the method. :)

File: krzeelzb/assignment-4/src/main/java/WorldMap.java (hunk 6 above)
Comment (krzeelzb) [reply krzeelzb/assignment-4/src/main/java/WorldMap.java]: changed to not static

File: krzeelzb/assignment-4/src/main/java/Island.java (hunk 9 above)
Comment (krzeelzb) [reply krzeelzb/assignment-4/src/main/java/Island.java]: changed

**Pull request merged**